### PR TITLE
New methods: path (115 B), pathEq (449 B), pathOr (176 B), pathSatisfies (177 B)

### PR DESCRIPTION
**Pull request by BabkinAleksandr**
[Ramda path](http://ramdajs.com/docs/#path)
[Ramda pathEq](http://ramdajs.com/docs/#pathEq)
[Ramda pathOr](http://ramdajs.com/docs/#pathOr)
[Ramda pathSatisfies](http://ramdajs.com/docs/#pathSatisfies)

### CODE REVIEW

File: lib/path/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function path(arr, obj) {
+  return arr.length > 1 && obj[arr[0]] !== undefined
+    ? path(arr.slice(1), obj[arr[0]])
```
Comment (Beraliv) [inline lib/path/index.js]: it's ineffective I suppose, because every iteration has a new allocation of array using `slice`

Comment (BabkinAleksandr) [reply lib/path/index.js]: Hmm, what's your proposal? Can't simply come up with another variant )

Comment (Beraliv) [reply lib/path/index.js]: Yeah, totally forgot to advise you to use `for(;;)` with `var state = obj` which saves current pointer to the result

**Comment by BabkinAleksandr** [comment]
Improved `path`. Added performance tests, and also added new generator function for nested objects.

**Pull request merged**